### PR TITLE
docs - awscc_iam_oidc_provider

### PR DESCRIPTION
**Pull request by naveen-amzn**
### Community Note

* Please vote on this pull request by adding a 👍 [reaction](https://blog.github.com/2016-03-10-add-reactions-to-pull-requests-issues-and-comments/) to the original pull request comment to help the community and maintainers prioritize this request
* Please do not leave "+1" or other comments that do not add relevant new information or questions, they generate extra noise for pull request followers and do not help prioritize the request
* The resources and data sources in this provider are generated from the CloudFormation schema, so they can only support the actions that the underlying schema supports. For this reason submitted bugs should be limited to defects in the generation and runtime code of the provider. Customizing behavior of the resource, or noting a gap in behavior are not valid bugs and should be submitted as enhancements to AWS via the CloudFormation Open Coverage Roadmap.

Relates OR Closes #0000

### CODE REVIEW

File: docs/resources/iam_oidc_provider.md [hunk 1]
```diff
@@ -10,7 +9,29 @@ description: |-
 
 Resource Type definition for AWS::IAM::OIDCProvider
 
-
+## Example Usage
+
+### Basic example
```
Comment (rahul-amzn) [inline docs/resources/iam_oidc_provider.md:14]: should we remove 'Basic example', as this includes single example?

File: docs/resources/iam_oidc_provider.md [hunk 2]
```diff
@@ -10,7 +9,29 @@ description: |-
...
+Description about the first example
```
Comment (rahul-amzn) [inline docs/resources/iam_oidc_provider.md:15]: should we update like: 

`To create IAM OpenID Connect provider with tags`

Comment (naveen-amzn) [reply docs/resources/iam_oidc_provider.md]: updated the code.

File: examples/resources/awscc_iam_oidc_provider/iam_oidc_provider.tf [hunk 3]
```diff
@@ -0,0 +1,17 @@
...
+    },
+    {
+      key   = "Environment"
+      value = "development"
+    },
+    { key   = "Modified By"
+      value = "AWSCC"
+  }]
+
+}
```
Comment (wellsiau-aws) [inline examples/resources/awscc_iam_oidc_provider/iam_oidc_provider.tf:17]: I suggest to use example with Terraform Cloud as the OIDC provider , basically adopting this example: https://github.com/hashicorp/terraform-dynamic-credentials-setup-examples/blob/main/aws/aws.tf

```suggestion
# Data source used to grab the TLS certificate for Terraform Cloud.
data "tls_certificate" "tfc_certificate" {
  url = "https://app.terraform.io"
}

resource "awscc_iam_oidc_provider" "this" {
  thumbprint_list = [data.tls_certificate.tfc_certificate.certificates[0].sha1_fingerprint]
  client_id_list  = ["aws.workload.identity"]
  url             = data.tls_certificate.tfc_certificate.url
  tags = [
    {
      key   = "Name"
      value = "IAM OIDC Provider"
    },
    {
      key   = "Environment"
      value = "Development"
    },
    { key   = "Modified By"
      value = "AWSCC"
    }
  ]
}
```

**Review by wellsiau-aws — CHANGES_REQUESTED** [review]

**Review by wellsiau-aws — APPROVED** [review]
LGTM

**Comment by marcosentino** [comment]
@naveen-amzn Thank you for the contribution!

**Pull request merged**